src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -146,6 +169,7 @@ protected internal virtual void OnWorkspaceFailed(WorkspaceDiagnostic diagnostic
     /// <summary>
     /// An event that is fired when a <see cref="Document"/> is opened in the editor.
     /// </summary>
+    [Obsolete("This member is obsolete. Use the RegisterDocumentOpenedHandler(Func<DocumentEventArgs, Task>) overload instead.", error: true)]
```

Comment:
we can likely `error: false` initially.  `error: true` will need more discussion :)

Reply:
I'm going to go ahead and remove these, exposing the new registration methods as public is needed before obsoleting these